src/ast/raw.ts

```diff
@@ -0,0 +1,15 @@
+export default class Raw {
+  readonly $$mdtype = 'Raw' as const;
+
+  static isRaw = (node: any): node is Raw => {
+    return !!(node?.$$mdtype === 'Raw');
+  };
+
+  content: string;
+  inline: boolean;
+
+  constructor(content: string, inline: boolean) {
```

Comment:
I think I might be missing something, but is there any reason we can't treat HTML as a scalar of some sort? And not have to introduce this new AST type?

Reply:
Conventional strings are [escaped](https://github.com/markdoc/markdoc/blob/115699172cbb824a6a8b39e3edec0a868e596aeb/src/renderers/html.ts#L26-L27) for security reasons—symbols like '<' and '>' are replaced with the equivalent HTML entity references. If we pass the raw HTML all the way through as a string, it's going to render literally instead of as HTML. Putting the raw content in a dedicated AST node makes it possible to distinguish it from regular text.

It's also important to capture and preserve whether it originated in a block or inline context because that may have implications for how you render it (e.g. if you are going to put it in a `dangerouslySetInnerHtml` attribute on an HTML element, you might want to use `div` or `span` depending on where the content originated)